DDG4/src/Geant4GeneratorWrapper.cpp

```diff
@@ -77,11 +77,7 @@ void Geant4GeneratorWrapper::operator()(G4Event* event)  {
   // Now generate the new interaction
   generator()->GeneratePrimaryVertex(event);
 
-  /// Add all the missing interactions (primary vertices) to the primary event record.
-  for(G4PrimaryVertex* gv=event->GetPrimaryVertex(); gv; gv=gv->GetNext())  {
-    if ( primaries.find(gv) == primaries.end() )   {
-      Geant4PrimaryInteraction* inter = createPrimary(m_mask, primaryMap, gv);
-      prim->add(m_mask, inter);
-    }
-  }
+  // Add all the missing interactions (primary vertices) to the primary event record.
+  Geant4PrimaryInteraction* inter = createPrimary(m_mask, primaryMap, primaries);
+  prim->add(m_mask, inter);
```

Comment:
To be added:
``` 
for(G4PrimaryVertex* gv=event->GetPrimaryVertex(); gv; gv=gv->GetNext())  {
    Geant4Vertex* v = createPrimary(gv);
   inter->vertices.emplace(v, gv);  
   for (G4PrimaryParticle *gp = gv->GetPrimary(); gp; gp = gp->GetNext() )
     collectPrimaries(pm, interaction, v, gp);
}
```

Reply:
This doesn't compile, but I think most of that is already done in createPrimary?

Though, we will discuss in person and see.